src/dash-table/components/ControlledTable/index.tsx

```diff
@@ -225,119 +248,183 @@ export default class ControlledTable extends PureComponent<ControlledTableProps>
+
+            if (!ths) {
+                return;
+            }
+
+            ths.forEach(this.clearCellWidth);
+        });
+    }
+
+    resizeFragmentCells = (
+        fragment: HTMLElement,
+        widths: number[]
+    ) => {
+        const lastRowOfCells = fragment.querySelectorAll<HTMLElement>('table.cell-table > tbody > tr:last-of-type > *');
+        if (!lastRowOfCells.length) {
             return;
         }
 
-        if (!force) {
-            setState({ forcedResizeOnly: true });
+        Array.from(
+            lastRowOfCells
+        ).forEach((c, i) => this.setCellWidth(c, widths[i]));
+
+        const firstThs = Array.from<HTMLElement>(fragment.querySelectorAll('table.cell-table > tbody > tr > th:first-of-type'));
+        const trOfThs = firstThs.map(th => th.parentElement);
+
+        trOfThs.forEach(tr => {
+            const ths = Array.from<HTMLElement>(tr?.children as any);
+
+            if (!ths) {
+                return;
+            }
+            if (ths.length === widths.length) {
+                ths.forEach((c, i) => this.setCellWidth(c, widths[i]));
+            } else {
+                ths.forEach(c => this.setCellWidth(c, 0));
+            }
+        });
+    }
+
+    resizeFragmentTable = (table: HTMLElement | null, width: string) => {
+        if (!table) {
+            return;
         }
 
-        this.updateStylesheet();
+        table.style.width = width;
+    }
 
-        getScrollbarWidth().then((scrollbarWidth: number) => setState({ scrollbarWidth }));
+    forceHandleResize = () => this.handleResize();
 
+    handleResizeIf = memoizeOne((..._: any[]) => {
         const { r0c0, r0c1, r1c0, r1c1 } = this.refs as { [key: string]: HTMLElement };
 
+        r0c1.style.marginLeft = '';
+        r1c1.style.marginLeft = '';
+        r0c0.style.width = '';
+        r1c0.style.width = '';
 
-        // Adjust [fixed columns/fixed rows combo] to fixed rows height
-        let trs = r0c1.querySelectorAll('tr');
-        Array.from(r0c0.querySelectorAll('tr')).forEach((tr, index) => {
-            const tr2 = trs[index];
+        const r0c0Table = r0c0.querySelector('table');
+        const r0c1Table = r0c1.querySelector('table');
+        const r1c0Table = r1c0.querySelector('table');
 
-            tr.style.height = `${tr2.clientHeight}px`;
-        });
+        if (r0c0Table) { r0c0Table.style.width = ''; }
+        if (r0c1Table) { r0c1Table.style.width = ''; }
+        if (r1c0Table) { r1c0Table.style.width = ''; }
 
-        // Adjust fixed columns headers to header's height
-        let trths = r1c1.querySelectorAll('tr > th:first-of-type');
-        Array.from(r1c0.querySelectorAll('tr > th:first-of-type')).forEach((th, index) => {
-            const tr2 = trths[index].parentElement as HTMLElement;
-            const tr = th.parentElement as HTMLElement;
+        this.resetFragmentCells(r0c0);
+        this.resetFragmentCells(r0c1);
+        this.resetFragmentCells(r1c0);
```

Comment:
Minor - skip reset for a table that doesn't exist? Also seems like this could be 🌴
```
[r0c0, r0c1, r1c0].forEach(rc=> {
    rcTable=...;
    if (rcTable) { rcTable.style.width = ''; this.resetFragmentCells(rc); }
});
```

Reply:
https://github.com/plotly/dash-table/pull/793/commits/575aa2c4bc838474bcfa251c8596897be6c76d17#diff-a6637ea69a08d602c311a574b4e4979aR340